Algorithm.Framework/Risk/TrailingStopRiskManagementModel.cs

```diff
@@ -53,37 +53,36 @@ public override IEnumerable<IPortfolioTarget> ManageRisk(QCAlgorithm algorithm,
                 // Remove if not invested
                 if (!security.Invested)
                 {
-                    if (_trailingHighs.ContainsKey(symbol))
+                    if (_trailing.ContainsKey(symbol))
                     {
-                        _trailingHighs.Remove(symbol);
+                        _trailing.Remove(symbol);
                     }
                     continue;
                 }
 
+                var profitPercent = security.Holdings.UnrealizedProfitPercent;
+
                 // Add newly invested securities
-                if (!_trailingHighs.ContainsKey(symbol))
+                if (!_trailing.ContainsKey(symbol))
                 {
-                    _trailingHighs.Add(symbol, security.Holdings.AveragePrice); // Set to average holding cost
+                    _trailing.Add(symbol, profitPercent); 
                     continue;
                 }
 
-                // Check for new highs and update - set to tradebar high
-                if (_trailingHighs[symbol] < security.High)
+                // Check for new high and update
+                if (_trailing[symbol] < profitPercent)
                 {
-                    _trailingHighs[symbol] = security.High;
+                    _trailing[symbol] = profitPercent;
```

Comment:
`_trailing[symbol] = profitPercent;` should probably start from 0 here? 

Suggest directly using `TryGetValue()` instead of 3 access: `ContainsKey` + `[symbol]` twice
Same for python 👍

Reply:
Hey @Martin-Molinero thanks for review, good point, have fixed it. Hope python version is done right, I am not a big expert lol) As for the unit test - should this be another regression algorithm? or regular test? Sorry didn't quite get it. Thanks,

Reply:
Hey! 
> Hey @Martin-Molinero thanks for review, good point, have fixed it. Hope python version is done right, I am not a big expert lol)

Looks good 💯 

> As for the unit test - should this be another regression algorithm? or regular test? Sorry didn't quite get it. Thanks,

Would be nice to add unit tests, example for another risk model at https://github.com/QuantConnect/Lean/blob/master/Tests/Algorithm/Framework/Risk/MaximumDrawdownPercentPerSecurityTests.cs

Reply:
Reminder: branch needs a rebase from master